tests/storage_test.go

```diff
@@ -67,71 +68,99 @@ var _ = Describe("[Serial]Storage", func() {
 	})
 
 	Describe("Starting a VirtualMachineInstance", func() {
-		var _pvName string
 		var vmi *v1.VirtualMachineInstance
-		var nfsInitialized bool
-
-		initNFS := func() string {
-			if !nfsInitialized {
-				_pvName = "test-nfs" + rand.String(48)
-				// Prepare a NFS backed PV
-				By("Starting an NFS POD")
-				os := string(cd.ContainerDiskAlpine)
-				nfsIP := tests.CreateNFSTargetPOD(os)
-				// create a new PV and PVC (PVs can't be reused)
-				By("create a new NFS PV and PVC")
-				tests.CreateNFSPvAndPvc(_pvName, "5Gi", nfsIP, os)
 
-				nfsInitialized = true
-			}
-			return _pvName
+		initNFS := func() *k8sv1.Pod {
+			tests.SkipNFSTestIfRunnigOnKindInfra()
```

Comment:
all over, if sriov doesnt run those tests, `SkipNFSTestIfRunnigOnKindInfra` can be removed please, unless you want to protect all future kind lanes.
we first added it because NFS didnt work on kind ipv6 lane, i dont know if we ever checked it on sriov lane (maybe its already skipped there by the ginko focus level)

Reply:
> I think it was explained that individual `afterEach` helps remove the boilerplate code that is needed. We depend on the framework to call the registered teardown functions instead of building this ourself in one block.

I dont expect or want us to create our own teardown fucntion (AfterEach equivalent if that what you mean)

> Why it is wrong to use the framework teardown per resource vs one teardown block for all (or several).

I assume that by "framework teardown" you mean "AfterEach"
Like I said in earlier comments, In general as a reviewer its harder to follow multiple `AfterEach` blocks per `Context` instead of single one, its also not common practice in those tests `Context`'s or whole kubevirt test suite.
In NFS PV tests there is no benefit from doing that due the the dependencies of the object you want to delete.

Reply:
At the current state sriov lane runs only sriov tests, and not deploying CDI operator

Reply:
This PR didn't add it, it is not in its scope to remove it.